src/main/java/dev/codesupport/web/common/security/JwtRequestFilter.java

```diff
@@ -39,71 +46,108 @@ public JwtRequestFilter(AuthenticationUserDetailsService userDetailsService) {
+            JsonWebToken jwToken = jsonWebTokenFactory.createToken(tokenString);
+
+            String username = jwToken.getUsername();
+
+            configureSpringSecurityContext(username, request);
+        } catch (InvalidTokenException e) {
+            // Doesn't matter, just log it for debugging and the service with return an unauthorized error.
+            if (log.isDebugEnabled()) {
+                log.debug("Failed to validate token", e);
             }
         }
     }
 
+    /**
+     * Configures the Spring Security Context
+     * <p>If not already configured, this method configures the Spring Security Context, which is
+     * used later by the framework for access checks, etc.</p>
+     *
+     * @param username The user to be configured in the security context
+     * @param request  The requested associated to the configuration
+     */
+    void configureSpringSecurityContext(String username, HttpServletRequest request) {
+        Assert.notNull(username, "username can not be null");
+
+        // If authentication is not already configured
+        if (SecurityContextHolder.getContext().getAuthentication() == null) {
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+
+            UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            usernamePasswordAuthenticationToken
+                    .setDetails(new WebAuthenticationDetailsSource().buildDetails(request));
+
+            SecurityContextHolder.getContext().setAuthentication(usernamePasswordAuthenticationToken);
+        }
+    }
+
     /**
      * Extract token from header.
      *
      * @param request Corresponding request to extract token from
      * @return The raw token ascii string
      */
     String getTokenFromRequest(HttpServletRequest request) {
-        final String requestTokenHeader = request.getHeader("Authorization");
+        final String requestAuthorizationHeader = request.getHeader("Authorization");
         String jwtToken;
 
         // Get the token from the header
-        if (StringUtils.startsWith(requestTokenHeader, "Bearer ")) {
-            jwtToken = requestTokenHeader.substring(7); //Removes "Bearer " prefix
-        } else { // Invalid authorization header
+        if (!StringUtils.isEmpty(requestAuthorizationHeader)) {
+            if (StringUtils.startsWith(requestAuthorizationHeader, "Bearer ")) {
+                jwtToken = requestAuthorizationHeader.substring(7); //Removes "Bearer " prefix
+            } else { // Invalid authorization header
+                jwtToken = null;
+                if (log.isDebugEnabled()) {
+                    logger.debug("Authorization header does not begin with Bearer ");
+                }
+            }
+        } else { // Missing authorization header
             jwtToken = null;
```

Comment:
See above comment.

Reply:
how does this relate to the above comment?